core/types/log.go

```diff
@@ -144,19 +144,16 @@ func (logs Logs) Filter(addrMap map[common.Address]struct{}, topics [][]common.H
 	return o
 }
 
-func (logs Logs) CointainTopics(addrMap map[common.Address]struct{}, topicsMap map[common.Hash]struct{}, maxLogs uint64) Logs {
+func (logs Logs) ContainingTopics(addrMap map[common.Address]struct{}, topicsMap map[common.Hash]struct{}, maxLogs uint64) Logs {
 	o := make(Logs, 0, len(logs))
 	var logCount uint64
-	logCount = 0
+
 	for _, v := range logs {
 		found := false
 
 		// check address if addrMap is not empty
-		if len(addrMap) != 0 {
-			if _, ok := addrMap[v.Address]; !ok {
-				// not there? skip this log
-				continue
-			}
+		if _, requested := addrMap[v.Address]; !requested && len(addrMap) > 0 {
```

Comment:
1. in golang usually variables of reading from map called `_, ok`: https://go.dev/blog/maps
2. `ok` is less meaningful than `requested`. But `_, ok := requestedAddrs[v.Addess]` is better. (in-general including data-structure into variable name is useless in typed languages).
3. if check `!ok` then no reason to check `len(addrMap) > 0`